cache/redis_cache.go

```diff
@@ -329,6 +341,22 @@ func (r *redisCache) Name() string {
 	return r.name
 }
 
+func (f *redisCache) checkAlive() {
+	for {
+		select {
+		case <-f.quitAliveCheck:
+			return
+		default:
+			f.alive = f.client.Ping(context.Background()).Err() == nil
+			time.Sleep(pingInterval)
+		}
+	}
+}
+
+func (f *redisCache) Alive() bool {
+	return f.alive
```

Comment:
While most likely it isn't an issue now (there is only one reader and one writer). There is a race condition on the alive boolean and an atomic variable would be better suited.

Reply:
This idea crossed my mind but I thought it wasn't necessary for now. I can work on something if needed